infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInfraModule.java

```diff
@@ -77,8 +78,11 @@ protected void configure() {
     volumesStrategies.addBinding(UNIQUE_STRATEGY).to(UniqueWorkspacePVCStrategy.class);
     bind(WorkspaceVolumesStrategy.class).toProvider(WorkspaceVolumeStrategyProvider.class);
 
-    MapBinder<String, ExternalServerExposerStrategy> ingressStrategies =
-        MapBinder.newMapBinder(binder(), String.class, ExternalServerExposerStrategy.class);
+    MapBinder<String, ExternalServerExposerStrategy<KubernetesEnvironment>> ingressStrategies =
+        MapBinder.newMapBinder(
+            binder(),
+            new TypeLiteral<String>() {},
```

Comment:
I am curious: do you really need to use `TypeLiteral` for String (that is not a generic type)?

Reply:
I need TypeLiteral for `ExternalServerExposerStrategy<KubernetesEnvironment>` and there is no binding method with mixed class and type literal arguments

Reply:
Ok makes sense thanks